lib/Dialect/Torch/Utils/Utils.cpp

```diff
@@ -333,3 +333,25 @@ bool Torch::isAssumingStrictSymbolicShapes(Block *block) {
   }
   return false;
 }
+
+void Torch::checkDefaultStrideHelper(PatternRewriter &rewriter,
+                                     SmallVector<Value> sizeListValues,
+                                     SmallVector<Value> strideListValues,
```

Comment:
You can use `SmallVector<OpFoldResult>` instead of values, and then use `rewriter.createOrFold` calls to create the ops down below and evaluate the assert at compile time when everything is static.

Reply:
Oh cool. I'll do that.

Reply:
@qedawkins so I tried changing those SmallVectors but there's undefined conversions from Value to OpFoldResult that make that and several other things in this function not work. I made as many things as I could as "createOrFold", but I'm not clear on how that gets handled exactly. I suppose one thing I could do is forcibly fail a test, get the ir, correct it, and then see if it folds away.